packages/server-wallet/src/chain-service/__chain-test__/chain-service.test.ts

```diff
@@ -198,6 +212,69 @@ describe('fundChannel', () => {
 });
 
 describe('registerChannel', () => {
+  it('dispatches a channel finalized event if the channel has been finalized BEFORE registering', async () => {
+    const channelId = randomChannelId();
+    const CHALLENGE_EXPIRE_TIME = 6000;
+    const FUTURE_TIME = 7000;
+
+    const tx = await testAdjudicator.functions.setChannelStorageHash(
+      channelId,
+      channelDataToChannelStorageHash({
+        turnNumRecord: 0,
+        finalizesAt: CHALLENGE_EXPIRE_TIME,
+      })
+    );
+    await tx.wait();
+    const channelFinalizedHandler = jest.fn();
+    const channelFinalizedPromise = new Promise<void>(resolve =>
+      chainService.registerChannel(channelId, [ethAssetHolderAddress], {
+        holdingUpdated: _.noop,
+        assetTransferred: _.noop,
+        channelFinalized: arg => {
+          channelFinalizedHandler(arg);
+          resolve();
+        },
+      })
+    );
+    await mineBlock(FUTURE_TIME);
+    await channelFinalizedPromise;
+    expect(channelFinalizedHandler).toHaveBeenCalledWith({channelId});
+  });
+
+  it('registers a channel in the finalizing channel list and fires an event when that channel is fainlized', async () => {
+    const channelId = randomChannelId();
+    const CURRENT_TIME = 5000;
+    const CHALLENGE_EXPIRE_TIME = 6000;
+    const FUTURE_TIME = 7000;
+
+    const tx = await testAdjudicator.setChannelStorageHash(
+      channelId,
+      channelDataToChannelStorageHash({
+        turnNumRecord: 0,
+        finalizesAt: CHALLENGE_EXPIRE_TIME,
+      })
+    );
+    await tx.wait();
+
+    const channelFinalizedHandler = jest.fn();
+    const channelFinalizedPromise = new Promise<void>(resolve =>
+      chainService.registerChannel(channelId, [ethAssetHolderAddress], {
+        holdingUpdated: _.noop,
+        assetTransferred: _.noop,
+        channelFinalized: arg => {
+          channelFinalizedHandler(arg);
+          resolve();
+        },
+      })
+    );
+
+    await mineBlock(CURRENT_TIME);
+    expect(channelFinalizedHandler).not.toHaveBeenCalled();
```

Comment:
I _think_ this check is not sufficient since it is possible that `channelFinalizedHandler` has not fired **yet**. Consider the following sequence of events:
1. current time block is mined.
1. The logic in this line is executed.
1. The `channelFinalizedHandler`fires.
1. future time block is mined.

That being said, I am not sure how to properly check that `channelFinalizedHander` did not fire after the first block is mined. We could insert a delay, but that slows down the testsuite.

Reply:
Yeah I agree. I couldn't think of a better way of doing it without introducing a delay. I think it's probably worthwhile adding a very short delay so the test is more likely to catch any issues.

Reply:
I've added 1 sec delay in d101ee4.